src/include/sof/audio/component.h

```diff
@@ -863,7 +865,8 @@ static inline struct comp_dev *comp_alloc(const struct comp_driver *drv,
 	 * Use uncached address everywhere to access components to rule out
 	 * multi-core failures. TODO: verify if cached alias may be used in some cases
 	 */
-	dev = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM, bytes);
+	dev = drv_heap_rzalloc(drv->drv_heap, SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM,
+				bytes);
```

Comment:
the size is expected to be passed to the firmware from topology https://github.com/thesofproject/linux/pull/5460

Reply:
module API already has a memory allocator API, we should repurpose it for user as ALL modules will be user.